### PR TITLE
Update USE_CASES.md with Whitelabel and Email Statistics information

### PR DESCRIPTION
**Pull request by alanunruh**
Added information to `USE_CASES.md`

- How to setup a domain whitelabel 

- How to view email statistics.

- Add Whitelabel and Email Statistics to Table of Contents

This resolves issue #204 .

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=205) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=205) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-ruby?pullRequest=205) it.</sub>

File: USE_CASES.md [hunk 1]
```diff
@@ -110,4 +110,16 @@ end
 puts response.status_code
 puts response.body
 puts response.headers
-```
\ No newline at end of file
+```
+
```
Comment (thinkingserious) [inline USE_CASES.md]: Please add entries into the [Table of Contents](https://github.com/sendgrid/sendgrid-ruby/blob/master/USE_CASES.md#table-of-contents) and we will be good to go :)

Comment (alanunruh) [reply USE_CASES.md]: Nice catch! PR has been updated.

**Comment by thinkingserious** [comment]
Hello @alanunruh,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by mbernier** [comment]
@alanunruh 
If you are interested, we would love the opportunity to talk to you about Hacktoberfest and our API libraries.

Our agenda would be: Explore what you liked and is there anything we can do to improve? 

You can grab a time [on my calendar](https://calendly.com/matt-bernier/hacktoberfest-follow-up) that works for you and we can have a chat on Google Hangout or Skype. If you prefer, you can email me using my GitHub username at my company’s domain.

Thank you so much,

Matt Bernier - @mbernier - SendGrid Developer Experience Product Manager
Elmer Thomas - @thinkingserious - SendGrid Developer Experience Engineer
@SendGrid

**Comment by alanunruh** [comment]
@mbernier I've sent you a date. I look forward to it!

**Pull request merged**